### PR TITLE
bf(S3C-3940): Prevent reporting of negative metrics

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

File: libV2/server/API/metrics/listMetrics.js [hunk 1]
```diff
@@ -16,6 +16,14 @@ const metricResponseKeys = {
     service: 'serviceName',
 };
 
+function positiveOrZero(value) {
+    if (value < 0) {
+        return 0;
+    }
+
+    return value;
+}
```
Comment (miniscruff) [inline libV2/server/API/metrics/listMetrics.js]: Can use math max https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max

Comment (tmacro) [reply libV2/server/API/metrics/listMetrics.js]: fancy

File: libV2/server/API/metrics/getStorage.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ async function getStorage(ctx, params) {
 
     ctx.results.statusCode = 200;
     ctx.results.body = {
-        storageUtilized,
+        storageUtilized: storageUtilized > 0 ? storageUtilized : 0,
```
Comment (miniscruff) [inline libV2/server/API/metrics/getStorage.js]: math max works here too

**Review by miniscruff — APPROVED** [review]
would like a test or two but looks good to me

**Review by anurag4DSB — APPROVED** [review]

File: libV2/server/API/metrics/getStorage.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ async function getStorage(ctx, params) {
 
     ctx.results.statusCode = 200;
     ctx.results.body = {
-        storageUtilized,
+        storageUtilized: Math.max(storageUtilized, 0),
```
Comment (anurag4DSB) [inline libV2/server/API/metrics/getStorage.js:57]: Can it go in negative as well?

**Review by jonathan-gramain — APPROVED** [review]

File: libV2/server/API/metrics/getStorage.js (hunk 3 above)
Comment (tmacro) [reply libV2/server/API/metrics/getStorage.js:57]: Technically all UTAPI metrics can go negative as it is, at its core, just a system of summing numbers that you give it. This means that if the `+` metrics are missed but the `-` ones aren't you get negatives.

**Pull request merged**